src/sources/soundsourceproxy.cpp

```diff
@@ -361,6 +361,12 @@ SoundSourceProxy::exportTrackMetadataBeforeSaving(Track* pTrack, UserSettingsPoi
                 pTrack->updateStreamInfoFromSource(
                         pAudioSource->getStreamInfo());
                 DEBUG_ASSERT(pTrack->hasStreamInfoFromSource());
+                // Make sure that the track file is closed after reading.
```

Comment:
This additional code should become obsolete with the fix I just proposed.

Reply:
This is still needed:
```
proxy.closeAudioSource();
```

Reply:
I would like to keep this as an hot fix solution with a minimum impact.

The rest is fixed in https://github.com/mixxxdj/mixxx/pull/4584

Reply:
I would prefer a proper fix here.

Reply:
I am not going to merge a hack with a wrong comment.

Reply:
Why do you think the comment is wrong? Add least it is working.

But I don't mind to merge your ideas.
Can you issue a PR against my branch? 

Thank you.

Reply:
The alternative would be to merge https://github.com/mixxxdj/mixxx/pull/4584 to this branch. This is IMHO the proper fix, but touches more code.

Reply:
I haven't reviewed #4584 yet and I cannot promise to do it soon.

Reply:
https://github.com/daschuer/mixxx/pull/72

Moreover the build failed on Linux with Clang due to a wrong preprocessor directive. Fixed.